azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java

```diff
@@ -52,6 +52,7 @@ public class TriggerManager extends EventHandler implements TriggerManagerAdapte
   private static final Logger logger = Logger.getLogger(TriggerManager.class);
   private static final Map<Integer, Trigger> triggerIdMap = new ConcurrentHashMap<>();
   private static final Set<Integer> removedTriggerIds = new HashSet<>();
+  private static final Set<Integer> backExecuteEnabledTriggerIds = new HashSet<>();
```

Comment:
I'm surprised all the Sets in this class are not "ConcurrentHashSet". The data structure is not thread-safe.

Reply:
Oh, i see it now. Thanks for your explanation!